### ISSUE TITLE
Resolve "Twisted: NEXT" from documentation

### ISSUE THREAD

**Issue by adiroiban** [twisted/twisted#12293]
It looks like our tooling arround "Twisted NEXT" can be improved 

I see a few placed in the docs where "Twsited NEXT" was not resolved to the actual version

https://docs.twisted.org/en/stable/core/howto/defer-intro.html#coroutines-with-async-await

We need to see what happend and how we can prevent this in the future.

I plan to work on this as part of `towncrier` https://github.com/twisted/towncrier/issues/579

As part of `towncrier check` for the release branch, we can have check to make sure the release marker is not found in the source code.

-------

Current result

> New in version Twisted: NEXT
>
> Coroutines can be passed to yield in code based on [inlineCallbacks](https://docs.twisted.org/en/stable/api/twisted.internet.defer.html#inlineCallbacks).

Expected result

> New in version Twisted 24.7.0
>
> Coroutines can be passed to yield in code based on [inlineCallbacks](https://docs.twisted.org/en/stable/api/twisted.internet.defer.html#inlineCallbacks).

**Comment by glyph** [comment]
When I run `python -m incremental.update --rc Twisted` it only updates `.py` files.  Has it *always* skipped `.rst` files?

**Comment by adiroiban** [comment]
Maybe it's only .py files

I don't see test coverage in `incremental` for RST files

https://github.com/twisted/incremental/blob/trunk/src/incremental/tests/test_update.py

so maybe something to consider for `incremental`.

**Comment by twm** [comment]
Yes, Incremental has always skipped files outside the package directory. I was planning add support for controlling what files it substitutes in as part of addressing [#58.](https://github.com/twisted/incremental/issues/58).

**Comment by glyph** [comment]
For the existing instances, I suggest that we just do a bit of `git blame` spelunking to get the right numbers in there manually?
